### PR TITLE
Add manual polygon edge type setting

### PR DESCRIPTION
**Pull request by GTcreyon**
# Description of changes
Adds an exported dictionary to manually set the edge type of terrain polygons.

### CODE REVIEW

File: classes/solid/terrain/pencil.gd [hunk 1]
```diff
@@ -273,23 +253,24 @@ func get_connected_lines_directional(lines, add_onto_blacklist, direction, poly,
 		var normal: Vector2 = vert.direction_to(next).tangent()
 		var angle: float = normal.angle_to(direction) / PI * 180
 		
-		if angle >= -root.max_deviation and angle <= root.max_deviation:
+		if (!type_list.has(ind) and angle >= -root.max_deviation and angle <= root.max_deviation) or (type_list.has(ind) and type_list[ind] == type_id):
```
Comment (jaschutte) [inline classes/solid/terrain/pencil.gd:256]: Possibly make a seperate function which returns a bool if all requirements are met, makes this part of the code more readable

File: classes/solid/terrain/pencil.gd [hunk 2]
```diff
@@ -299,44 +280,45 @@ func get_connected_lines_blacklist(lines, blacklist, poly, start):
...
+		latest_index = get_connected_lines_directional(list, type_list, root.up_direction, poly, latest_index, 1)
 		if list.size() >= 2:
 			draw_top_from_connected_lines(list)
 	
 	# Do the bottom layer
 	latest_index = 0
 	while latest_index != null:
 		var list = []
-		latest_index = get_connected_lines_directional(list, blacklist, root.down_direction, poly, latest_index)
+		latest_index = get_connected_lines_directional(list, type_list, root.down_direction, poly, latest_index, 2)
```
Comment (jaschutte) [inline classes/solid/terrain/pencil.gd:302]: Use enums rather than a raw number? (Aka, type_id being an enum and not just 2)

**Review by jaschutte — COMMENTED** [review]
A few readability suggestions, otherwise perfectly fine code

**Comment by GTcreyon** [comment]
This can be reviewed now.

**Review by jaschutte — APPROVED** [review]
No issues found

**Pull request merged**